subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1161,6 +1161,10 @@ static int ztls_socket(int family, int type, int proto)
 	z_object_recycle(ctx);
 #endif
 
+	if (proto == IPPROTO_TCP) {
```

Comment:
It looks like there would be issue after the next tls allocation few lines below `if (ctx->tls == NULL) {`. If we get to that branch, the ctx would have ref count of 2, and the ctx would not be freed in that case. Looks like the ref should be incremented just before `z_finalize_fd()` call.

Reply:
As commented earlier, it seems that `net_context_put` will handle dereferencing, but there's no harm moving it below, perhaps it's cleaner this way.